backend/core/src/main/java/org/sonarsource/sonarlint/core/fs/ClientFile.java

```diff
@@ -123,6 +126,11 @@ public void setClean() {
     this.clientProvidedContent = null;
   }
 
+  public boolean isSonarlintConfigurationFile() {
```

Comment:
Add a comment about why we take `.sonarlint/*.json` files into account. Basically, the best path is only to have one, but there might be exceptions:
- SLVS saves Connected Mode configuration by solution name as JSON
- Users might want to have multiple ones (e.g., one for production, one for testing)